api/src/main/java/org/apache/iceberg/util/CharSequenceWrapper.java

```diff
@@ -54,7 +54,7 @@ public boolean equals(Object other) {
     CharSequenceWrapper that = (CharSequenceWrapper) other;
 
     if (wrapped instanceof String && that.wrapped instanceof String) {
-      return wrapped.equals(that.wrapped);
+      return wrapped.toString().contentEquals(that.wrapped);
```

Comment:
see 
[https://errorprone.info/bugpattern/UndefinedEquals](https://errorprone.info/bugpattern/UndefinedEquals) check For CharSequence

Reply:
Thanks for looking into this @ajantha-bhat, but there is a fair chance that this was intentional for performance reasons.

Reply:
I see. We can suppress the warning in that case then. 
@aokolnychyi: Thoughts?